lib/interfaces/include/BMSDriverGroup.h

```diff
@@ -0,0 +1,332 @@
+     */
+    void start_GPIO_ADC_conversion();
+
+    /* -------------------- PRINT DATA FUNCTIONS -------------------- */
+
+    /**
+     * Prints all the cell voltage information
+     * Every cell voltage, the minimum, maximum, and locations of those
+     */
+    void print_voltage_data();
+
+    /**
+     * @post resets the max, min data holders to outside bound
+     */
+    void reset_voltage_data();
+
+    /**
+     * Prints all the GPIO information
+     * Every temperature, humidity, the maximum temperature, and locations of those
+     */
+    void print_GPIO_data();
+
+    /**
+     * @post resets the max, min data holders to outside bound
+     */
+    void reset_GPIO_data();
+
+    /* -------------------- SETTER FUNCTIONS -------------------- */
+
+    /**
+     * Sets chip select for this IC
+     * @param integer input chip select
+     */
+    void set_chip_select(int ic_index, int cs) { this->chip_select = cs; }
+
+    /**
+     * ADDRESSES: Addresses only pertain to LTC6811-2s which are encoded in hardware through 4 bits
+     * These addresses need to be coded into the firmware so that address commands are properly sent
+     * This function is a setter for the address
+     * @param integer input address
+     */
+    void set_address(int *addr) { this->address = addr; }
+
+    /* -------------------- GETTER FUNCTIONS -------------------- */
+
+    /**
+     * Generates a Packet Error Code
+     * @pre PEC table is already generated –> pec15Table has data generated in it
+     * @post returns an unsigned 16 bit integer used to check data to and from the device
+     * @param data data we are writing PEC for, could be Command Code or Buffer data
+     * @param length length of data
+     * @return unsigned 16 bit PEC, array of uint8_t of length 2
+     */
+    uint8_t *calculate_specific_PEC(uint8_t *data, int length);
+
+    /**
+     * Generates a formmatted 2 byte array for the Command bytes
+     * @return unsigned 8 bit array of length 2
+     */
+    uint8_t *generate_formatted_CMD(CMD_CODES_e command, int ic_index);
+
+    /**
+     * Generates Command and PEC as one byte array of length 4: CMD0, CMD1, PEC0, PEC1
+     * @return unsigned 8 bit, length 4
+     */
+    uint8_t *generate_CMD_PEC(CMD_CODES_e command);
+
+    /**
+     * @return total thermistor temperature
+     *
+     */
+    float get_total_thermistor_temperature() { return total_thermistor_temps; }
+
+    /**
+     * @return total thermistor temperature
+     */
+    float get_total_board_temperature() { return total_board_temps; }
+
+    /**
+     * @return total voltage
+     */
+    uint16_t get_total_voltage() { return total_voltage; }
+
+    /**
+     * @return a usable command address of the specific LTC6811-2 chip to send command to
+     */
+    uint8_t get_address(int ic_index) { return address[ic_index]; }
+
+    /**
+     * @return IC_data container
+     */
+    IC_Buffers_s get_IC_buffer() { return *this->IC_buffer; }
```

Comment:
no need to have these getter functions if you are returning voltages / temps and humidities via the read functions